ibm/service/vpc/resource_ibm_is_virtual_network_interface_floating_ip_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright IBM Corp. 2023 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package vpc_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	acc "github.com/IBM-Cloud/terraform-provider-ibm/ibm/acctest"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+)
+
+func TestAccIBMIsVirtualNetworkInterfaceFloatingIPBasic(t *testing.T) {
+	var conf vpcv1.FloatingIPReference
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMIsVirtualNetworkInterfaceFloatingIPDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMIsVirtualNetworkInterfaceFloatingIPConfigBasic(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMIsVirtualNetworkInterfaceFloatingIPExists("ibm_is_floating_ip.is_floating_ip", conf),
+				),
+			},
+		},
+	})
+}
+
+func TestAccIBMIsVirtualNetworkInterfaceFloatingIPAllArgs(t *testing.T) {
+	var conf vpcv1.FloatingIPReference
+	name := fmt.Sprintf("tf_name_%d", acctest.RandIntRange(10, 100))
+	nameUpdate := fmt.Sprintf("tf_name_%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMIsVirtualNetworkInterfaceFloatingIPDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMIsVirtualNetworkInterfaceFloatingIPConfig(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMIsVirtualNetworkInterfaceFloatingIPExists("ibm_is_floating_ip.is_floating_ip", conf),
+					resource.TestCheckResourceAttr("ibm_is_floating_ip.is_floating_ip", "name", name),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCheckIBMIsVirtualNetworkInterfaceFloatingIPConfig(nameUpdate),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("ibm_is_floating_ip.is_floating_ip", "name", nameUpdate),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "ibm_is_floating_ip.is_floating_ip",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckIBMIsVirtualNetworkInterfaceFloatingIPConfigBasic() string {
+	return fmt.Sprintf(`
+		resource "ibm_is_floating_ip" "is_floating_ip_instance" {
+		}
+	`)
+}
+
+func testAccCheckIBMIsVirtualNetworkInterfaceFloatingIPConfig(name string) string {
+	return fmt.Sprintf(`
+
+		resource "ibm_is_floating_ip" "fip" {
+			name = "%s"
+			zone = "%s"
+		}
+		resource "ibm_is_virtual_network_interface" "vni" {
+			allow_ip_spoofing = %s
+			auto_delete = %s
+			enable_infrastructure_nat = %s
+			ips {
+				address = "192.168.3.4"
```

Comment:
address and id should be mutually exclusive

Reply:
code changed